lib/nerve.rb

```diff
@@ -45,15 +43,41 @@ def run
       end
 
       begin
-        sleep
-      rescue StandardError => e
+        loop do
+          # Check that watcher threads are still alive, auto-remediate if they
+          # are not. Sometimes zookeeper flakes out or connections are lost to
+          # remote datacenter zookeeper clusters, failing is not an option
+          relaunch = []
+          @watchers.each do |name, watcher_thread|
+            unless watcher_thread.alive?
+              log.warn "nerve: watcher #{name} not alive, attempting to relaunch"
```

Comment:
this log line should end up in the relaunch section, since otherwise the logging will look like all watchers relaunched successfully; if the first relaunch goes poorly, it'll look like the last one did